packages/discovery-provider/src/queries/get_user_purchased_content.py

```diff
@@ -0,0 +1,118 @@
+import logging
+from typing import TypedDict
+
+from sqlalchemy import and_, or_  # pylint: disable=C0302
+
+from src.models.playlists.playlist import Playlist
+from src.models.social.repost import RepostType
+from src.models.social.save import SaveType
+from src.models.tracks.track import Track
+from src.models.users.usdc_purchase import USDCPurchase
+from src.queries.query_helpers import (
+    add_query_pagination,
+    populate_playlist_metadata,
+    populate_track_metadata,
+)
+from src.utils import helpers
+from src.utils.db_session import get_db_read_replica
+
+logger = logging.getLogger(__name__)
+
+
+class GetUserPurchasedContentArgs(TypedDict):
+    current_user_id: int
+    user_id: int
+    limit: int
+    offset: int
+
+
+def _get_user_purchasable_content(session, args: GetUserPurchasedContentArgs):
+    """Fetch tracks owned by the given user that have been remixed by other users."""
+
+    user_id = args.get("user_id")
+
+    query = (
+        session.query(USDCPurchase, Track, Playlist)
+        .outerjoin(
+            Track,
+            and_(
+                USDCPurchase.content_type == "track",
+                USDCPurchase.content_id == Track.track_id,
+                Track.is_current == True,
```

Comment:
We really need a mixin or some kind of way to enforce this logic is the same in all places we query for tracks/playlists :-/